### PR TITLE
[CINFRA-673] Multi shard snapshot transfer

### PR DESCRIPTION
**Pull request by mpoeter**
### Scope & Purpose

Prepare for transferring multiple shards during snapshot transfer.

- [x] :pizza: New feature

### Checklist

- [x] Tests
  - [x] C++ **Unit tests**

#### Related Information

- [x] Jira ticket: https://arangodb.atlassian.net/browse/CINFRA-673

### CODE REVIEW

**Review by maierlars — APPROVED** [review]
LGTM

File: arangod/Replication2/StateMachines/Document/CollectionReader.cpp [hunk 1]
```diff
@@ -28,31 +28,66 @@
...
+auto ReplicationTransaction::options() -> transaction::Options {
   transaction::Options options;
   options.requiresReplication = false;
+  options.fillBlockCache = false;  // TODO - check whether this is a good idea
+  return options;
+}
+
+auto ReplicationTransaction::addCollection(LogicalCollection const& collection)
+    -> Result {
+  return transaction::Methods::addCollection(collection.id(), collection.name(),
```
Comment (apetenchea) [inline arangod/Replication2/StateMachines/Document/CollectionReader.cpp:49]: I had a quick look at the Jenkins failure, and this seems to be the culprit. Unfortunately, we cannot call `addCollection` after the transaction has status `CREATED`, which happens after it is constructed. We might have to create a transaction for each shard, which clearly would be less elegant than what you were trying to achieve here, or find some other workaround.

**Review by apetenchea — CHANGES_REQUESTED** [review]

File: arangod/Replication2/StateMachines/Document/CollectionReader.cpp (hunk 1 above)
Comment (mpoeter) [reply arangod/Replication2/StateMachines/Document/CollectionReader.cpp:49]: We definitely should be able to add readonly collections to an already startet trx. Will look into this!

Comment (apetenchea) [reply arangod/Replication2/StateMachines/Document/CollectionReader.cpp:49]: When we try to do that, an exception is thrown in `Methods.cpp:3001`: "cannot add collection to a previously started top-level transaction". But, that can work with `addCollectionAtRuntime`. I don't understand the inner workings of these methods either, just tried it out and it works.

**Review by apetenchea — APPROVED** [review]
LGTM!

**Pull request merged**